exec/java-exec/src/main/java/org/apache/drill/exec/vector/complex/MapUtility.java

```diff
@@ -228,11 +228,301 @@ public static void writeToMapFromReader(FieldReader fieldReader, BaseWriter.MapW
+          } else {
+            fieldReader.copyAsValue(mapWriter.bit(fieldName));
+          }
+          break;
+        case VARCHAR:
+          if (repeated) {
+            fieldReader.copyAsValue(mapWriter.list(fieldName).varChar());
+          } else {
+            fieldReader.copyAsValue(mapWriter.varChar(fieldName));
+          }
+          break;
+        case VARBINARY:
+          if (repeated) {
+            fieldReader.copyAsValue(mapWriter.list(fieldName).varBinary());
+          } else {
+            fieldReader.copyAsValue(mapWriter.varBinary(fieldName));
+          }
+          break;
+        case MAP:
+          if (valueMajorType.getMode() == TypeProtos.DataMode.REPEATED) {
+            fieldReader.copyAsValue(mapWriter.list(fieldName).map());
+          } else {
+            fieldReader.copyAsValue(mapWriter.map(fieldName));
+          }
+          break;
+        case LIST:
+          fieldReader.copyAsValue(mapWriter.list(fieldName).list());
+          break;
+        default:
+          throw new DrillRuntimeException(String.format(caller
+              + " does not support input of type: %s", valueMinorType));
+      }
+    } catch (ClassCastException e) {
+      final MaterializedField field = fieldReader.getField();
+      throw new DrillRuntimeException(String.format(caller + TYPE_MISMATCH_ERROR, field.getName(), field.getType()));
+    }
+  }
+
+  public static void writeToListFromReader(FieldReader fieldReader, BaseWriter.ListWriter listWriter, String caller) {
+    try {
+      MajorType valueMajorType = fieldReader.getType();
+      MinorType valueMinorType = valueMajorType.getMinorType();
+      boolean repeated = false;
+
+      if (valueMajorType.getMode() == TypeProtos.DataMode.REPEATED) {
+        repeated = true;
+      }
+
+      switch (valueMinorType) {
+        case TINYINT:
+          fieldReader.copyAsValue(listWriter.tinyInt());
+          break;
+        case SMALLINT:
+          fieldReader.copyAsValue(listWriter.smallInt());
+          break;
+        case BIGINT:
+          fieldReader.copyAsValue(listWriter.bigInt());
+          break;
+        case INT:
+          fieldReader.copyAsValue(listWriter.integer());
+          break;
+        case UINT1:
+          fieldReader.copyAsValue(listWriter.uInt1());
+          break;
+        case UINT2:
+          fieldReader.copyAsValue(listWriter.uInt2());
+          break;
+        case UINT4:
+          fieldReader.copyAsValue(listWriter.uInt4());
+          break;
+        case UINT8:
+          fieldReader.copyAsValue(listWriter.uInt8());
+          break;
+        case DECIMAL9:
+          fieldReader.copyAsValue(listWriter.decimal9());
+          break;
+        case DECIMAL18:
+          fieldReader.copyAsValue(listWriter.decimal18());
+          break;
+        case DECIMAL28SPARSE:
+          fieldReader.copyAsValue(listWriter.decimal28Sparse());
+          break;
+        case DECIMAL38SPARSE:
```

Comment:
`VARDECIMAL` type should be added also here.

Reply:
Done